### PR TITLE
Update proto definition and dependencies.

### PR DESCRIPTION
**Pull request by GPSnoopy**
### TL;DR
Update to latest dependencies. Similar to PR #2.

### Details
In order to be able to use `arrow2` and `arrow-format` with the latest `tonic` and `prost` crates, I had to patch the repository as following (the old files generated with `prost` are not compatible with these new versions).

* Generate flight files from latest proto definition.
* Update `prost` dependency to 0.10.x.
* Update `tonic` dependency to 0.7.x.
* Update `crate` version to 0.4.0 to match crate.io version and `arrow2` dependency constraints.
* Run `cargo fmt --all`

Regarding the first point, I used the latest proto definition file from `arrow` github. There are no fundamental changes, only clarifications around comments.

I expect that the crate version should be bumped to 0.5.0 before releasing this crate (i.e. not part of this PR).

### CODE REVIEW

**Comment by jorgecarleitao** [comment]
Thanks a lot, makes a lot of sense! Will bump this one and bump it in arrow2 so that we have everything aligned 👍
